### PR TITLE
[new release] styled-ppx (0.56.0)

### PR DESCRIPTION
**Pull request by davesnx**
Type-safe styled components for ReScript and Melange

- Project page: <a href="https://styled-ppx.vercel.app">https://styled-ppx.vercel.app</a>

## 0.56.0

- Improvement for locations in both code-gen and error reporting (davesnx/styled-ppx#456) by @davesnx
- Support css min and max functions (davesnx/styled-ppx#411) by @lubegasimon
- Update docs (davesnx/styled-ppx#457) by @zakybilfagih
- Native support for styled.{{tag}} (davesnx/styled-ppx#461) by @zakybilfagih
- background-clip: text support by @davesnx
- Fix linear-gradient and radial-gradient (davesnx/styled-ppx#464) by @davesnx
- Rename emotion-hash into murmur2 and remove public testing cli by @davesnx
- Use server-reason-react from opam by @davesnx

### CODE REVIEW

File: packages/styled-ppx/styled-ppx.0.56.0/opam [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  "menhir" {>= "20220210"}
+  "ppx_deriving" {>= "5.0"}
+  "ppx_deriving_yojson" {>= "3.7.0"}
+  "ppxlib" {>= "0.27.0"}
+  "sedlex" {>= "3.2"}
+  "melange" {>= "3.0.0"}
+  "server-reason-react"
+  "reason-react" {>= "0.14.0"}
+  "alcotest" {with-test}
+  "reason-react-ppx" {with-dev-setup}
```
Comment (mseri) [inline packages/styled-ppx/styled-ppx.0.56.0/opam:23]:
```suggestion
  "conf-npm" {with-test}
  "reason-react-ppx" {with-dev-setup}
```
this should make node available

**Comment by samoht** [comment]
Something is going wrong with `packages/ppx/test/snapshot/rescript/rescript-bs-config-missing-jsx-version.t/run.t`. Is this expected? Should we disallow the tests for this package?

**Comment by davesnx** [comment]
Let me disable the rescript tests since it requires rescript to exist, will fix it and re-push

**Comment by davesnx** [comment]
I disabled those tests, and I'm getting a few errors that I'm not able to understand.

[extras,riscv64-ocaml-5.1,styled-ppx.0.56.0,tests](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/185ce8b48200146a35344b4fe72c0aa33354cc22/variant/extras,riscv64-ocaml-5.1,styled-ppx.0.56.0,tests) -> Looks like a problem with conf-npm?


[extras,s390x-ocaml-5.1,styled-ppx.0.56.0](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/185ce8b48200146a35344b4fe72c0aa33354cc22/variant/extras,s390x-ocaml-5.1,styled-ppx.0.56.0)

```
# File "packages/reason-css-parser/lib/.reason_css_parser.objs/native/_unknown_", line 1, characters 0-0:
# (cd _build/default && /home/opam/.opam/5.1/bin/ocamlopt.opt -w -40 -g -I packages/reason-css-parser/lib/.reason_css_parser.objs/byte -I packages/reason-css-parser/lib/.reason_css_parser.objs/native -I /home/opam/.opam/5.1/lib/ocaml/compiler-libs -I /home/opam/.opam/5.1/lib/ppx_deriving/runtime -I /home/opam/.opam/5.1/lib/sedlex -I packages/parser/lib/.styled_ppx_css_parser.objs/byte -I packages/parser/lib/.styled_ppx_css_parser.objs/native -intf-suffix .ml -no-alias-deps -open Reason_css_parser__ -o packages/reason-css-parser/lib/.reason_css_parser.objs/native/reason_css_parser__Parser.cmx -c -impl packages/reason-css-parser/lib/Parser.re.pp.ml)
# >> Fatal error: camlReason_css_parser__Parser.entry: function too complex, cannot complete register allocation
```

[freebsd,freebsd-ocaml-5.1-amd64,styled-ppx.0.56.0,tests](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/185ce8b48200146a35344b4fe72c0aa33354cc22/variant/freebsd,freebsd-ocaml-5.1-amd64,styled-ppx.0.56.0,tests)

```
#    $ ./compare.sh ""
# -  Hashes match: 0
# +  /tmp/dune_cram_8a4f44_.hash.t/1.sh: ./compare.sh: not found
```

Looks like ./compare.sh can't be found

**Comment by mseri** [comment]
I don;t know how to resolve the errors, I think we can merge and if some specific user encounters them they can help us fix them

**Pull request merged**